sdks/java/harness/src/main/java/org/apache/beam/fn/harness/Caches.java

```diff
@@ -166,6 +165,10 @@ public int weigh(Object key, WeightedValue<Object> value) {
                             return (int) size;
                           }
                         })
+                    // The maximum size of an entry in the cache is maxWeight / concurrencyLevel
+                    // which is why we set the concurrency level to 1. See
+                    // https://github.com/google/guava/issues/3462 for further details.
+                    .concurrencyLevel(1)
```

Comment:
Would be good to verify this setting doesn't add additional contention and add that justification to the comment.

If it is worse (are worker threads modifying this directly?) then perhaps it would be better to keep the concurrency level and just update option comment about maximum element size or log what the maximum size for element can be.

Reply:
I have a benchmark that I've been meaning to push up after this change that compared with and without cross bundle caching. I used it to check and it looks like there is no noticeable difference.

With the concurrency level at the default (4):
```
Result "org.apache.beam.fn.harness.jmh.ProcessBundleBenchmark.testStateWithCaching":
  4452.734 ±(99.9%) 60.263 ops/s [Average]
  (min, avg, max) = (4227.619, 4452.734, 4550.040), stdev = 80.449
  CI (99.9%): [4392.471, 4512.997] (assumes normal distribution)
```

With the concurrent level at the 1:
```
Result "org.apache.beam.fn.harness.jmh.ProcessBundleBenchmark.testStateWithCaching":
  4445.516 ±(99.9%) 41.809 ops/s [Average]
  (min, avg, max) = (4345.263, 4445.516, 4536.817), stdev = 55.814
  CI (99.9%): [4403.707, 4487.325] (assumes normal distribution)
```

Reply:
State caching benchmark in https://github.com/apache/beam/pull/16542

Reply:
Thanks!